content/blog/open-sourcing-workshops-on-demand-part4-managing-the-backend.md

```diff
@@ -0,0 +1,242 @@
+---
+title: "Open Sourcing Workshops-on-Demand part 4: Manage the Backend"
+date: 2023-07-18T09:28:35.992Z
+author: Frederic Passeron
+authorimage: /img/frederic-passeron-hpedev-192.jpg
+disable: false
+tags:
+  - " opensource"
+  - workshops-on-demand
+---
+I﻿n previous articles of this series dedicated to the [open sourcing of our Workshops-on-Demand project](https://developer.hpe.com/blog/willing-to-build-up-your-own-workshops-on-demand-infrastructure/), I covered the reasons why we open sourced  the project, how we did it. I also explained in details how you could install your own Workshops-on-Demand backend server. I also took the time to detail the automation that was hosted on this backend server. Today, I plan to describe to you the management of this backend server. What is often refered to as day2 operations. 
+
+O﻿nce up and running, the backend server main purpose is to deliver workshops-on-Demand. But to do so, it may require updates, upgrades, new kernels for the Jupyterhub server. If new workshops are created, this means new jinja templates for related workshops' scripts (`create<WKSHP>.sh`, `cleanup<WKSHP>.sh`, `reset<WKSHP>.sh` among others). This also means new variables files. And obviously, these templates and variables will need to be taken into account by scripts and notebooks. Some tasks handle all of this. And we will see now how.
+
+#### B﻿ackend server management:
+
+If you take a look at the file structure of the `wod-backend` directory, you will discover that we did our best to sort things properly depending on their relationship to system  or workshops.
+
+##### C﻿ontent of the backend server:
+
+S﻿imple tree view of the wod-backend directory:
+
+![](/img/wod-blogserie2-tree1.png "Tree view of wod-backend directory")
+
+T﻿he `ansible` folder contains all the necessary playbooks and variables files to support the main functions of the backend server. It provides playbooks for a minimal installation of the server or appliances, setup of the servers (backend, frontend, api-db),of appliances or workshops as well as maintenance tasks.
+
+A﻿t the root of this directory can be found:
+
+`C﻿heck*.yml playbooks`: These playbooks are used to perform checks on the different systems. These checks ensure that this a compliant WoD system by checking Firewall rules and many other things. We will see this a bit later in more details.
+
+`C﻿opy_folder.yml`: This is historically one of very first playbook we used and therefore a very important one. It performs the necessary actions to deploy, personnalize (by substituting ansible variables) the selected notebook to the appropriate student home folder.
```

Comment:
Modify this section "This is historically..." to:
"Historically, this is one of the very first playbooks the team used, and therefore, it is very important to me."

Change the second sentence to remove the comma and use the word "and" instead so it reads:
"It performs the necessary actions to deploy and personnalize (by substituting Ansible variables) the selected notebook to the appropriate student home folder." (note capitalization of "Ansible")